webhooks/virtualmachine/mutation/virtualmachine_mutator.go

```diff
@@ -310,6 +312,37 @@ func SetDefaultPowerState(
 	return false
 }
 
+// SetDefaultBiosUUID sets a default bios uuid for a new VM.
+// If CloudInit is the Bootstrap method, CloudInit InstanceID is also set to BiosUUID.
+// Return true if a default bios uuid was set, otherwise false.
+func SetDefaultBiosUUID(
+	ctx *context.WebhookRequestContext,
+	client client.Client,
+	vm *vmopv1.VirtualMachine) bool {
+
+	var wasMutated bool
+
+	if vm.Spec.BiosUUID == "" {
+		// Default to a Random (Version 4) UUID.
+		// This is the same UUID flavor/version used by Kubernetes and preferred by vSphere.
+		vm.Spec.BiosUUID = uuid.New().String()
+		wasMutated = true
+	}
+
+	if vm.Spec.Bootstrap != nil && vm.Spec.Bootstrap.CloudInit != nil {
```

Comment:
Would this also handle all the cases and be a bit more readable?
```go
if vm.Annotations == nil {
    vm.Annotations = map[string]string{}
}
                                                                                                                                                                                                                                                    	if vm.Spec.Bootstrap != nil && vm.Spec.Boot
if vm.Annotations[vmopv1.InstanceIDAnnotation] != vm.Spec.BiosUUID {
    vm.Annotations[vmopv1.InstanceIDAnnotation] = vm.Spec.BiosUUID
    wasMutated = true
}

```

Reply:
I'd meant to only create the map if nil, but realized after your comment I missed checking for nil map above:
```go
id := vm.Annotations[vmopv1.InstanceIDAnnotation]
```
This code may also go away after we resolve the cloud-init.UID discussion